src/linker/Linker/LinkContext.cs

```diff
@@ -308,7 +308,18 @@ public void RegisterAssembly (AssemblyDefinition assembly)
 		{
 			if (SeenFirstTime (assembly)) {
 				SafeReadSymbols (assembly);
-				Annotations.SetAction (assembly, CalculateAssemblyAction (assembly));
+				Annotations.SetAction (assembly,
+					IsCPPCLIAssembly (assembly.MainModule) ? AssemblyAction.Copy : CalculateAssemblyAction (assembly));
+			}
+
+			static bool IsCPPCLIAssembly (ModuleDefinition module)
+			{
+				foreach (var type in module.Types)
+					if (type.Namespace == "<CppImplementationDetails>" ||
+						type.Namespace == "<CrtImplementationDetails>")
+						return true;
+
+				return false;
```

Comment:
Why not move this into the `CalculateAssemblyAction` - that name actually suggests it would do things like this (unlike `RegisterAssembly).